types.go

```diff
@@ -131,3 +132,50 @@ type PrivateSectorInfo struct {
 type AllocationManager interface {
 	Free()
 }
+
+func DeDuplicatePrivateSectorInfo(ctx context.Context, sortPrivSectors *SortedPrivateSectorInfo) (SortedPrivateSectorInfo, error) {
+	var newSortPrivSectors SortedPrivateSectorInfo
+	newSortPrivSectors.f = make([]PrivateSectorInfo, 0)
+	for i := range sortPrivSectors.f {
+		if !ExistSector(newSortPrivSectors.f, sortPrivSectors.f[i].SectorNumber) {
+			newSortPrivSectors.f = append(newSortPrivSectors.f, sortPrivSectors.f[i])
+		}
+	}
+
+	new_sector_len := len(newSortPrivSectors.f)
+	if new_sector_len < 2 {
+		return newSortPrivSectors, nil
+	}
+
+	for i := 0; i < new_sector_len; i++ {
+		flag := false
+		for j := 0; j < new_sector_len-i-1; j++ {
+			if newSortPrivSectors.f[j].SectorNumber > newSortPrivSectors.f[j+1].SectorNumber {
+				newSortPrivSectors.f[j], newSortPrivSectors.f[j+1] = newSortPrivSectors.f[j+1], newSortPrivSectors.f[j]
+				flag = true
+			}
+		}
+		if !flag {
+			break
+		}
```

Comment:
The above code seems to sort sector infos by SectorNumber, but the original code sorts them by SealedCID, why the change?

Also if you want to sort, please use `sort.Slice`. See types.go#93

Reply:
@Kubuxu According to the logic before splitting,Split calculation results need to be sorted by SectorNumber
![企业微信截图_16377349048939](https://user-images.githubusercontent.com/77282562/143188987-2022d5b9-c9c2-44a1-a9b0-3cf5e817446d.png)

Reply:
> 

@Kubuxu Any suggestions for modification?

Reply:
> 

Because the members of the SortedProvateSectorInfo structure are private and cannot be operated in lotus, we think the DeDuplicatePrivateSectorInfo interface is necessary, or are there any other implementation suggestions? @Kubuxu

Reply:
Won't passing sectors deduplicated here to `NewSortedPrivateSectorInfo` sort them correctly? If not, can we change `NewSortedPrivateSectorInfo` to sort correctly?

Reply:
@magik6k We can change NewSortedPrivateSectorInfo to sort correctly, If It without affecting the logic of other functions

Reply:
@magik6k Modification has been submitted, please review